File: forkjoin-apikit-example/src/main/java/org/forkjoin/apikit/example/api/BaseApiController.java
package org.forkjoin.apikit.example.api;

import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;

import java.lang.Object;
import org.forkjoin.apikit.core.*;
import org.forkjoin.apikit.example.form.TestForm;
import org.forkjoin.apikit.example.model.*;
import org.springframework.web.bind.annotation.PathVariable;

/**
 * @author  zuoge85 on 15/6/11.
 */
@Api
@RestController
@RequestMapping(value = "v1")
public class BaseApiController {

	@RequestMapping(value = "base", method = RequestMethod.POST)
	public Result<TestObject<User[]>[]> create(@Valid TestForm<User> testForm, Object accountName) throws Exception {
		return Result.createSuccess();
	}

	@RequestMapping(value = "base/{id}", method = {RequestMethod.POST})
	public Result<Void> get(@PathVariable String id, Object user) throws Exception {
		return Result.createSuccess();
	}

	@RequestMapping(value = "baseUrl/", method = {RequestMethod.GET, RequestMethod.POST})
	public Result<User> create(@Valid User user, Object accountName) throws Exception {
		return Result.createSuccess();
	}
}

File: forkjoin-apikit-example/src/test/js/api/BaseApiController.js

import AbstractApi from './../AbstractApi'

import requestGroupImpi from './../RequestGroupImpi'


class BaseApiController extends AbstractApi {

    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>base</b>
    * <ul>
    * <li><b>Form:</b>TestFormcreate</li>
    * <li><b>Model:</b> TestObject[]</li>
    * <li>需要登录</li>
    * </ul>
    * </div>
    * @see TestObject[]
    * @see TestForm

     */
    create = (testForm) => {
        let _path = null;
        return super._request("baseApiController", "POST", "base", _path, testForm);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>base/{id}</b>
    * <ul>
    * <li><b>PathVariable:</b> string id</li>
    * <li><b>Model:</b> void</li>
    * <li>需要登录</li>
    * </ul>
    * </div>
    * @see string

     */
    get = (id) => {
        let _path = {};
        _path["id"] = id;
        return super._request("baseApiController", "POST", "base/{id}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>baseUrl/</b>
    * <ul>
    * <li><b>Form:</b>Usercreate</li>
    * <li><b>Model:</b> User</li>
    * <li>需要登录</li>
    * </ul>
    * </div>
    * @see User
    * @see User

     */
    create = (user) => {
        let _path = null;
        return super._request("baseApiController", "GET", "baseUrl/", _path, user);
    }

}

export { BaseApiController };
const baseApiController = new BaseApiController();
baseApiController._init(requestGroupImpi);
export default baseApiController;

